lib/HTML/FormBuilder/Select.pm

```diff
@@ -63,8 +65,12 @@ has name => (
 has options => (
     is  => 'rw',
     isa => sub {
-        die "$_[0] is not ArrayRef[HashRef[Any]]"
-            unless (ref($_[0]) eq 'ARRAY' && ref($_[0][0]) eq 'HASH');
+        my $arg = $_[0];
+        my $msg = Dumper($arg) . " is not ArrayRef[Maybe[HashRef]]";
+        die $msg if !ref($arg) || ref($arg) ne 'ARRAY';
+        for my $elem (@$arg) {
+            die $msg if (!ref($elem) || ref($elem) ne 'HASH');
+        }
```

Comment:
Ok I think we need to think about how we want to handle this.

Personally I think one should be able to pass in:
1.  An empty arrayref or
2.  just not specify at all and assign later.

Once we start allowing explicit undefs etc. then I think preconditions are going to be harder to manage.

Does this sound like a good approach?